Hyundai: Add Car Port for 2022 Kia Niro EV

sunnyhaibin opened the pull request:
**Checklist**
- [x] added to RELEASES, CARS
- [x] test route added to [test_routes.py](https://github.com/sunnyhaibin/openpilot-1/blob/94ddfc7163f2f0cb5dbc9eea7df69606e4db0ab8/selfdrive/test/test_routes.py#L92)
- [x] route with openpilot: `c011e480f2430f67|2021-10-28--11-09-58`

Thanks to community Niro EV owner Kolibabas.

sunnyhaibin committed:
Add car port: Kia Niro EV 2022

sunnyhaibin committed:
Update RELEASES.md and CARS.md

sunnyhaibin committed:
Add test route

kolibabas commented:
Not sure why it would fail on Route Not Found. I have the route preserved and set to public.

c011e480f2430f67|2021-10-28--11-09-58 

https://useradmin.comma.ai/?onebox=c011e480f2430f67%7C2021-10-28--11-09-58

adeebshihadeh commented on docs/CARS.md:
```diff
@@ -120,6 +120,7 @@ Community Maintained Cars and Features
 | Kia       | Forte 2018-21                 | SCC + LKAS        | Stock            | 0mph               | 0mph         |
 | Kia       | K5 2021                       | SCC + LFA         | Stock            | 0mph               | 0mph         |
 | Kia       | Niro EV 2019-21               | SCC + LKAS        | Stock            | 0mph               | 0mph         |
+| Kia       | Niro EV 2022                  | SCC + LKAS        | Stock            | 0mph               | 0mph         |
```
```suggestion
| Kia       | Niro EV 2019-22               | SCC + LKAS        | Stock            | 0mph               | 0mph         |
```

sunnyhaibin committed:
Update CARS.md

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh commented:
It's unclear that this needs a new car port. The existing Niro EVs never fingerprint correctly with the FW we have, so I suspect they're having the same issue as this 2022 Niro (i.e. the esp fw isn't from a Niro).

adeebshihadeh committed:
boardd: resolve TODO, still need that for USB errors

adeebshihadeh committed:
comma pedal: rename create_gas_command -> create_gas_interceptor_command

deanlee committed:
wifiManager/get_response: pass response by const reference (#22997)

adeebshihadeh committed:
boardd: set thread names

adeebshihadeh committed:
boardd: add note about slow disconnected panda check

adeebshihadeh committed:
Honda Bosch cleanup (#22999)

jyoung8607 committed:
VW MQB: Longitudinal prereqs (#23004)

adeebshihadeh committed:
perfetto (#22998)

Co-authored-by: Comma Device <device@comma.ai>

ErichMoraga committed:
Add Lexus RC support (#22941)

* Add f/w firmware for Lexus RC under LEXUS_IS

`@ph33rl355#0529`  2020 Lexus RC F Track Edition  DongleID/route 32696cea52831b02|2021-11-16--23-12-02

Doesn't seem to need a full port, per user feedback...
https://discord.com/channels/469524606043160576/524327905937850394/910400788319326229

* Update CARS.md

* Separated Lexus RC from IS... still using IS DBC

* match lexus IS for now

* Update wheelbase to match published specs

I'm hesitant to touch `mass`, even though I know it's different from the IS.

* Add TOYOTA.LEXUS_RC test route

Co-authored-by: Willem Melching <willem.melching@gmail.com>
Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh committed:
back to 0.8.12

adeebshihadeh committed:
boardd: small cleanup (#23005)

adeebshihadeh committed:
camerad: lower YUV vipc buffer count to 40 on tici

deanlee committed:
UI: fix cameraview crash after going offroad in settings window (#23009)

ErichMoraga committed:
Add several missing CHRH f/w (#23002)

`Hookler#6341` 2017 C-HR DongleID/route 9de14f6c78d29e91|2021-11-22--03-48-14
His car was previously FP'ing over FPv1 on OP 0.8.1, and he didn't realize it, until I verified it.  Relevant FPv1 was pruned from newer code.

pd0wm committed:
don't build internal kalman objects on device (#23011)

deanlee committed:
networking: remove unused #inculde<QMovie> (#23016)

geohot committed:
thermald: track power usage / pmic temps (#23013)

Co-authored-by: Comma Device <device@comma.ai>
Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

deanlee committed:
loggerd: split loggerd.cc into three files

deanlee committed:
panda: add unit tests for usb protocol (pack/unpack) (#22955)

* prepare for unit tests

* add to selfdrive_tests.yaml

* test header

* test chunk count

* rename test function

* continue

* don't check chunks count

* test recv_can

* continue

* small cleanup

* merge master

* cleanup

* rename functions

* test different packet size

* fix operator precedence problem

* refactor unpack_can_buffer

* cleanup test

* cleanup unpack_can_buffer

* add test for multiple pandas

* rename to test_panda

* restore test_boardd

* rename to test_boardd_usbprotocol

* fix typo

* bus_offset = [0,4]

* change src

* use USBPACKET_MAX_SIZE

adeebshihadeh committed:
Bring back Accord 2021 + 2021 Hybrid (#23021)

Co-authored-by: Chris Souers <csouers@gmail.com>

adeebshihadeh committed:
can_replay: warning & clone cmd if jungle lib is missing

robbederks committed:
Typo in Quectel FCC ID

adeebshihadeh committed:
mazda: capitalize CX9 like all others cars

adeebshihadeh committed:
update refs for new mazda name

deanlee committed:
loggerd: add test case for sync_encoders (#23020)

adeebshihadeh committed:
process replay: add cx9 route to migration list

nuwandavek committed:
locationd: Fix Nav localization reliability (#22959)

* modify reset logic

* remove debug statements

* use ecef pos and vel covariances during reset

* reset orientations initialized to 0,0,GPSbearing

* refactor nav fix

* add fake gps observations to control ecef pos and ecef vel std

* replace fake_P with individual fake cov

* set gps mode flag

* add gps invalid flag names

* update refs

* more accurate gps accuracy check + update refs

deanlee committed:
boardd: const reference pandaStates to avoid copy (#23030)

ZwX1616 committed:
test: Replay both models on C2/C3 (#23019)

* add dcam stream

* delete model_test

* both in jenkins

* update spinner

* get filename fn

* they are not random names

* update route

* new ref commit

* temporarily resize tmp

* clean up unlog send

* need calib 0

* revert device

* cleanup

* arg

Co-authored-by: Comma Device <device@comma.ai>

jyoung8607 committed:
VW MQB: Add FW for 2013 Volkswagen Golf (#23036)

grekiki2 committed:
replace list comprehensions with generators (#23037)

eFiniLan committed:
UI: minor changes to make it translation friendly (#23032)

pd0wm committed:
navd: render map into VisionIPC (#22800)

* navd: render simple map

* render route

* offscreen rendering

* cleanup

* more cleanup

* render into visionIPC

* rename class

* split position update from route update

* stop broadcast if not active

* gate vipc server behind flag

* add python library

* faster

* no vipc from python

* put behind extras

* only send when loaded

* add glFlush just to be sure

* cleanup settings into helper function

* function ordering

* broadcast thumbnails

* put behind param

* adjust zoom level

* add route to python bindings

* revert that freq change

* add logging if map rendering is enabled

* use rlogs if available

* bump cereal

deanlee committed:
qt/settings: use default parameter for params.get (#23042)

deanlee committed:
qt/settings: reuse params (#23041)

deanlee committed:
framereader: replace swscale with libyuv,  reduce cpu usage by half  (#22992)

* use libyuv

* cleanup

deanlee committed:
loggerd: add test case for trigger_rotate (#23038)

* test rotate

* remove global LoggerdState

sshane committed:
Clean up planner files (#23031)

* clean up planner files

clean up planner files

* fix plant instance

pd0wm committed:
bump cereal

deanlee committed:
ui/device: change last_brightness from float to int (#23029)

jwolffe committed:
CARLA: send YUV and RGB over visionipc from bridge.py (#23012)

* CARLA: send YUV and RGB over visionipc

* CARLA: send YUV and RGB over visionipc-fix pipfile

* CARLA: send YUV and RGB over visionipc-Camerad class

* relock pipfile

* small bridge cleanup

* use tici camera resolution

* update vof

* HUD_SETTING has no counter

* no loggerd

Co-authored-by: jwolffe <wolffja@gmail.com>
Co-authored-by: Willem Melching <willem.melching@gmail.com>

geohot committed:
boardd: be careful with paren order in assert

geohot committed:
Revert "boardd: be careful with paren order in assert"

This reverts commit 50cd8588a27834bfd01d95ed452cf8ac247cb584.

briskspirit committed:
Extra check for tail_size length (#23047)

adeebshihadeh committed:
camerad: remove camera frame stream (#23045)

* camerad: remove camera frame stream

* fix release files

pd0wm committed:
sim: improve carla performance (#23046)

deanlee committed:
framereader: fix crash after fallback to cpu decoding (#23055)

adeebshihadeh committed:
script to apply patch from PR

adeebshihadeh committed:
remove dead + duplicate python code

adeebshihadeh committed:
juggle: add demo flag and improve README

ErichMoraga committed:
Fix comment typo: steerRation -> steerRatio (#23058)

deanlee committed:
replay: refactor http download (#23052)

* refactor http download

* use chunk_size instead of parts

testcase:set chunksize to 5mb

* use template

space

* cleanup

* remove unused include

* check buffer overfllow

* simplify print download speed

deanlee committed:
DevicePanel: split long constructor into multiple functions (#23043)

* begin refactor

* split long constructor to multiple functions

* one connection to signal

* merge master

deanlee committed:
refactor httprequest: emit single signals (#23039)

* refactor httprequest

* Trigger Build

deanlee committed:
boardd: new function sync_time (#23033)

* sync_time

* add direction

* Update selfdrive/boardd/boardd.cc

Co-authored-by: Willem Melching <willem.melching@gmail.com>

* enum class

* rename

* caps

* lambda get_time_str

* Revert "lambda get_time_str"

This reverts commit 5eb6e19c5130b28963e9555f6c73835ac30d817d.

* static

Co-authored-by: Willem Melching <willem.melching@gmail.com>

sshane committed:
longitudinal: only apply overshoot prevention when braking (#22986)

* only if braking is desired

* use v_target to determine stopping

* more clear?

deanlee committed:
cameraview: fix frameskip (#23022)

* fix frameskip

* sync upload->render

* cleanup

* comment

* rename to latest_texture_id

adeebshihadeh committed:
manager: kill procs in parallel (#23010)

* manager: kill procs in parallel

* oops block

deanlee committed:
framereader: refactor seeking (#22926)

* refactor seeking

* ckeck packets.empty()

Co-authored-by: Willem Melching <willem.melching@gmail.com>

pd0wm committed:
Clear loggerd locks once on boot (#23060)

* add test for clear_locks

* move to bootlog

* simplify test

deanlee committed:
logreader: support reading from corrupt log (#23050)

* catch exception outside loop

* print decompress error

* add test case for corrupt log

* fix decompressbz2 stuck if log is corrupt

* recovered from corrupt data

* add output

* ass space

* std::endl

* override load(), load from buffer

* override FrameReader::load to load from the buffer

* replace NULL with nullptr

* fix test case for corrupt log

* Trigger Build

* check bzerror too

Co-authored-by: Willem Melching <willem.melching@gmail.com>

deanlee committed:
ui: singleton networkAccessManager (#22994)

pd0wm committed:
Add LastPowerDropDetected and create bootlog before clearing params (#23062)

* Add LastControlledShutdown and create bootlog before clearing params

* update param name

* sorting

pd0wm committed:
Do not run code on version.py import (#23063)

* Do not run code on version.py import

* fix athena

Commit:
fix pmic temp scale on comma two

adeebshihadeh committed:
logreader: remove log path print

haraschax committed:
Longcontrol : pid error should be 0 when plan is followed (#23066)

* v_pid error should be 0 when plan is followed

* update ref

briskspirit committed:
boardd: split usbprotocol unittest and fix paren order in data length assert (#23065)

adeebshihadeh committed:
script to verify release and staging branches are the same

adeebshihadeh committed:
add devel

deanlee committed:
rename yuv streams (#23071)

* rename yuv streams

* bump cereal

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

nuwandavek committed:
Mesh3D: Add accelerometer bias to loc_kf (#22880)

* add accel-bias to mesh3d

* remove acc scale

adeebshihadeh committed:
jenkins: build releases in parallel (#23070)

grekiki2 committed:
Replace list by generator (#23077)

ErichMoraga committed:
docs/CARS: Add 2021 C-HR (#23078)

`@eFini#8426` confirmed...
https://discord.com/channels/469524606043160576/524327905937850394/914925655228031017 [my inquiry]
https://discord.com/channels/469524606043160576/524327905937850394/915131655679791144 [his confirmation]

Still stuck w/ stock longitudinal, despite being TSS2.5.

He's been using it for many months, as seen in one of his early posts that came to Adeeb's attention for route analysis...
https://discord.com/channels/469524606043160576/524327905937850394/827038344742699008 [`88551059c417a545|2021-03-29--23-34-50`]

haraschax committed:
Toyota pedal rewrite (#23067)

* pedal redo

* add offset to compensate for creep and windbrake

* offset in standard units

* wrong size

* better creep values

* update ref

nuwandavek committed:
add a note in loc_kf (#23082)

deanlee committed:
soundd: fix test case (#23075)

* init sound_stats

* send deviceState msg

adeebshihadeh committed:
increase scons cache size for CI devices

adeebshihadeh committed:
process replay: set full env in cpp replay

adeebshihadeh committed:
New sounds (#22652)

* new engage/disengage + amp config

* first family

* cleanup audible alerts

* tici isn't special

* fix up debug cycle alerts

* these were better

* extend range

* use distracted sound

* log scaling

* getting closer

* slightly louder

* prompt

* update tests

* update refs

* fix c2 test

* resolve todo

* adjust tolerance

* revert for now

* should work

Co-authored-by: Comma Device <device@comma.ai>

adeebshihadeh committed:
add new sounds to release notes

adeebshihadeh committed:
test car models: ignore more fake buses

adeebshihadeh committed:
jenkins: make agent per-stage

nuwandavek committed:
locationd: cleanup (#23088)

* remove dead code, avoid indexing by value in locationd

* remove dead states in live_kf

pd0wm committed:
ci: improve valgrind test robustness (#23092)

haraschax committed:
Less sensitive torque saturated warning (#23003)

* Less sensetive torque saturated warning

* better comment

adeebshihadeh committed:
Honda: fix controls mismatch on Alt Nidecs (#23089)

* Honda: fix controls mismatch on Ridgeline

* bump panda

VirtuallyChris committed:
Update Videos in readme (#23094)

adeebshihadeh committed:
test car models: add test for addr checks (#23093)

* test car models: add test for addr checks

* revert that

* add checks

* ugh broken cars

* which seg

* f

* move that

adeebshihadeh committed:
update opendbc

adeebshihadeh committed:
Honda: fix address checks for Alt Nidec (#23099)

* Honda: fix address checks for Alt Nidec

* bump panda

briskspirit committed:
bump panda

mitchellgoffpc committed:
New model: d2053554-6094-43ce-943a-7bee3f43458d/950 (#23097)

* New model: d2053554-6094-43ce-943a-7bee3f43458d/950

* Updated model replay ref

* Revert "Updated model replay ref"

This reverts commit e83e0a1adc79dd20232b74932de202d858ebb485.

* New refs (again)

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

haraschax committed:
Set accel trajectory to a_ego when not engaged (#23100)

* Set accel trajectory to a_ego when not engaged

* update ref

adeebshihadeh committed:
Newer sounds (#23091)

* new prompt sound

* padding

* prompt distracted

* debug

* fix spelling

* lower min volume

* better volume at low speeds

* revert debug

* revert that

* fix tests

* update refs

Co-authored-by: Comma Device <device@comma.ai>

MyNameIsCosmo committed:
keyboard: replace backtick with apostrophe (#23104)

* Replace existing symbol with apostrophe

Without `'`, I cannot type my WiFi password... though it might be time
to rotate it.

* Replace extra bactic with apostrophe

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

pd0wm committed:
manager.py use get_short_branch() for GitBranch param

ErichMoraga committed:
Add several missing LEXUS_ES_TSS2 firmwares (#23098)

`@TechGuy#9141`  2021 Lexus ES 350 ICE  DongleID/route  f9238e084316715b|2021-12-01--19-26-20

adeebshihadeh committed:
MultiLogIterator: default wraparound to false

deanlee committed:
CameraView: use unique_ptr for CameraView::program (#23096)

adeebshihadeh committed:
soundd: max volume at 20m/s

haraschax committed:
add tss2 highlander to tss2 tune (#23107)

haraschax committed:
Human inspired follow distance (#22937)

* Add cost for creep

* more reasonable cost

* seems better than before

* science tune

* more chill

* closer follow

* stopping takes some leeway

* better cruise

* needed to be cumsum all along

* jerk not even needed

* matches better

* 6 is too much

* add back

* a little extra buffer is good for badly tuned cars

* new refs

* refs again

flightguy2 committed:
2019 Lexus NX FW versions (#23106)

deanlee committed:
panda: refactor can_recv (#22977)

* refactor

* use macro

* rebase master

* fix incorrect chunk_len

* cleanup

* type fix

* lines

* while

Co-authored-by: Igor Biletksyy <bs@privacy.im>

deanlee committed:
soundd: use setLoopCount(0) to stop repeating sound (#23076)

* use setLoopCount(0) to stop sound

* print default backend name

adeebshihadeh committed:
soundd: fix test for repeating soundsgitt status (#23110)

Co-authored-by: Comma Device <device@comma.ai>

haraschax committed:
Reaction buffer on follow distance (#23112)

* Reaction time buffer

* new ref

pd0wm committed:
add CarParams.wheelSpeedFactor (#23079)

* add CarParams.wheelSpeedFactor

* Fudge speed for Lexus RX

* should have been the TSS2 model

* bump cereal

* refactor into function

* update refs

pd0wm committed:
add desired angle to log (#23115)

* add desired angle to log

* update ref

* bump cereal

VirtuallyChris committed:
Honda Pilot fw (#23119)

haraschax committed:
Long policy: strictly squared stopped equivalence function (#23118)

* new ref

* never let lead stopped equivalence go down

* no overbraking

* stricter again

* cleanup

* updated refs

briskspirit committed:
len_to_dlc make static (#23120)

cydia2020 committed:
Toyota: remove unused ACCEL_HYST_GAP (#23121)

* fix typo in comments

* remove hyst gap from values.py

grekiki2 committed:
Python: Replace more lists with generators (#23116)

* Replace lists with generators v2

* Replace set with {}

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

* Replace more set() with {}

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh committed:
registration: remove key generation (#23111)

* registration: remove key generation

* add test

jyoung8607 committed:
VW MQB: Audi Q3 Mk2 (#23128)

adeebshihadeh committed:
jenkins: improve overall robustness (#23117)

* test onroad: more reliable log finding

* Update selfdrive/test/test_onroad.py

* write continue.sh in device setup

* clone

* try different delimeter

* try this instead

* fix keys path

* executable

* remove orphaned lock

deanlee committed:
move struct LogCameraInfo into loggerd.h (#23129)

adeebshihadeh committed:
updated: move update alerts out of thermald (#23028)

* updated: move update alerts out of thermald

* cleanup last update time

* set on startup

adeebshihadeh committed:
thermald: only go offroad on falling edge of ignition (#23130)

* thermald: only go offroad on falling edge of ignition

* and

* fix taht

khoi committed:
Add 2021 Toyoto Corolla Cross Non-Hybrid (Thailand version) (#23123)

deanlee committed:
UI: move onroad hud to Qt (#23040)

* onroad hud

* improve boudingrect, same as nanvg

* QRect

* Update selfdrive/ui/paint.cc

* stacked with nvgWindow

* cleanup ui.h

* rename to road_view_layout

* no lambda

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

sunnyhaibin committed:
HKG: Add FW for 2022 Kia K5 (#23131)

deanlee committed:
OnroadHud: update engageability and DM icons at 2Hz (#23134)

Commit:
fix sound test

adeebshihadeh committed:
Honda: use new Bosch LKAS_HUD messages (#23000)

* Honda: use new Bosch LKAS_HUD messages

* enum value

* better name

* flip that

adeebshihadeh committed:
remove ford fingerprints (#23136)

deanlee committed:
UI: draw header gradient in OnroadHud (#23137)

pd0wm committed:
nav: open map on new route without valid OpenGL context (#23140)

deanlee committed:
LogCameraInfo: remove unused variables (#23142)

jwolffe committed:
PC: fix onnxruntime excessive CPU (#23051)

* onnxruntime-gpu two threads

* relock pipfile

Co-authored-by: jwolffe <wolffja@gmail.com>
Co-authored-by: Willem Melching <willem.melching@gmail.com>

geohot committed:
add yy to gitignore

adeebshihadeh committed:
tici: use powersave CPU governor while offroad (#23146)

* tici: use powersave CPU governor while offroad

* fix path

deanlee committed:
UI: move rest of NVG to Qt (#23133)

* move ui_draw_world to qt

* draw world if world_objects_visible is true

* cleanup

* cleanup includes

* continue

* cleanup

* cleanup

* enable  multisampling on all devices

adeebshihadeh committed:
Remove NanoVG (#23147)

Commit:
fix UI build on comma two

adeebshihadeh committed:
fix startup test

haraschax committed:
add a change cost

adeebshihadeh committed:
loggerd: don't let a single service clog up the main thread

deanlee committed:
panda: refactor can_send (#22981)

* refactor

* merge master

* fix

* move  down

* space

* merge master

* remove struct PacketWriter, use function

* Update selfdrive/boardd/panda.cc

less violations

* Update selfdrive/boardd/panda.cc

comment

* Update selfdrive/boardd/panda.cc

add comment to counter

Co-authored-by: Igor Biletskyy <briskspirit@users.noreply.github.com>

adeebshihadeh committed:
increase loggerd per-cycle service limit to 200

haraschax committed:
long planner: no reset when starting (#23149)

* long planner: no reset when starting

* let go to 2m/s

* update ref

pd0wm committed:
Remove Raw Upload toggle (#23143)

* Remove Raw Upload toggle

* fix tests

deanlee committed:
DevicePanel: fix dangling reference to params (#23154)

VirtuallyChris committed:
Kia Niro EV fw (#23159)

JMPZ11 committed:
ui.h: add missing optional header include (#23160)

adeebshihadeh committed:
whole new family (#23157)

* family two

* more volume

* repeating dm sound

* update tests

haraschax committed:
Cruise speed: up to 90mph max (#23161)

* Cruise speed: up to 90mph max

* update ref

haraschax committed:
Add TSS2 prius to TSS2 long tunes

adeebshihadeh committed:
loggerd: fix delayed exit from main thread

adeebshihadeh committed:
onroad test fixes (#23164)

adeebshihadeh committed:
tici: higher cpu freq while offroad (#23151)

robbederks committed:
add check for restoring almanac, and clear like suggested in the datasheet (#23153)

Co-authored-by: Comma Device <device@comma.ai>

jwooning committed:
nav: athena destination with name and details (#23155)

pd0wm committed:
0.8.11-release FW batch (#23167)

* 0.8.11-release FW batch

* those toyota also dont always return esp

* attempt to clean up 2018 & 2021 accord

deanlee committed:
locationd_thread: fix possible dangling reference to params (#23156)

deanlee committed:
dmonitoring/DriverView: get crop_rect & transform by video size (#23014)

pd0wm committed:
Revert "dmonitoring/DriverView: get crop_rect & transform by video size (#23014)"

This reverts commit 1af715d6a20a1a7c5d11e85bf9420b2dab2334cc.

Metalbird1997 committed:
add fingerprint for European 2018 Toyota C-HR Hybrid (#23171)

mitchellgoffpc committed:
New model: 224877e7-dfcf-4b28-9482-371655918fe1/950 (#23170)

* New model: 224877e7-dfcf-4b28-9482-371655918fe1/950

* Updated model replay ref

adeebshihadeh committed:
UI: remove burn in prevention (#23173)

adeebshihadeh committed:
Merge 2021 Accords into 2018 Accords (#23174)

* Merge 2021 Accords into 2018 Accords

* last of them

haraschax committed:
Long mpc: fix xva policy for sim

haraschax committed:
XVA long policy: prev_a becomes numpy anyway (#23175)

adeebshihadeh committed:
tests for agnos manifest (#23176)

haraschax committed:
fix max speed comments

adeebshihadeh committed:
tici: lower offroad temp setpoint (#23177)

* tici: lower offroad temp setpoint

* Update selfdrive/thermald/thermald.py

adeebshihadeh committed:
tici: lower min volume

adeebshihadeh committed:
AGNOS 3 (#23172)

* AGNOS 3

* always forget to do this

* update

* release manifest

pd0wm committed:
dmonitoring/DriverView: get crop_rect & transform by video size (#23169)

* dmonitoring/DriverView: get crop_rect & transform by video size (#23014)

* keep rounding the same

Co-authored-by: Dean Lee <deanlee3@gmail.com>

deanlee committed:
limit style scope (#23181)

adeebshihadeh committed:
not separate model ports

adeebshihadeh committed:
alerts: add test for creation delay

adeebshihadeh committed:
revert to old red alert sound for now (#23182)

mitchellgoffpc committed:
Added sbigmodel frame transformations

adeebshihadeh committed:
alert text simplification (#23186)

* alert text simplification

* little more

* user triggered

* little more

* less annoying ldw

* Update selfdrive/controls/lib/events.py

* update refs

* revert

adeebshihadeh committed:
alerts: handle min duration properly (#23191)

* alerts: handle min duration properly

* add active

* tests

* cleanup test

* update refs

adeebshihadeh committed:
fix missing time import

adeebshihadeh committed:
updated: handle execptions while setting params

adeebshihadeh committed:
update release notes

adeebshihadeh committed:
add honda bosch hud msgs to release notes

haraschax committed:
Update RELEASES.md

geohot committed:
RELEASES.md: prevention is a bold statement

adeebshihadeh committed:
Revert "soundd: use setLoopCount(0) to stop repeating sound (#23076)"

This reverts commit 35885745f1b4fcaafee2ed7276de491e6df018d2.

adeebshihadeh committed:
alerts: escalate soft disable alert near disengagement (#23198)

* alerts: escalate soft disable alert if resulted in disengagement

* show 0.5s before disengagement

gregjhogan committed:
model output struct for metadata (#23139)

* model output struct for metadata

* better max element search

* ModelDataRawDesireProb.to_array()

* eliminate some copies

* not worth messing with softmax now

* remove unused includes

* more cleanup

* no longer pointers

* better variable name

* fix recurrent state

* improve variable name

* fix OUTPUT_SIZE and NET_OUTPUT_SIZE

adeebshihadeh committed:
increase cruise mismatch threshold

adeebshihadeh committed:
update refs

deanlee committed:
util: move all functions into util namespace (#23203)

deanlee committed:
RawLogger: fix wrong pts (#23205)

deanlee committed:
Use std::unique_ptr to manage memory (#23168)

* use unique_ptr

* trigger ci

* move down loggerdstate

deanlee committed:
commonmodel.h: move global model size into class ModelFrame (#23193)

* move to class ModelFrame

* remove defines in dmonitoring as well

Co-authored-by: Willem Melching <willem.melching@gmail.com>

deanlee committed:
common_process_driver_camera: same parameters as process_thread_cb (#23202)

deanlee committed:
replay: use std::unique_ptr to manage memory, reuse events buffer. (#23049)

* reuse events buffer

* Trigger Build

deanlee committed:
LapConv: remove hardcoded FULL_STRIDE_X (#23200)

* fix wroing stride_x

* remove hardcoded rgb_stride

deanlee committed:
Qt/SetingWindow: faster initialization and painting (#21764)

* reduce init time from ~520ms to ~350ms

* faster init toggle

setfixedheight in ctor

delay init TogglesPanel,reduce init time to ~180ms

cleanup

* precise style, reduce to ~100ms

* revert 'faster toggle

* fix horizontal_line

* fix background

* cleanup

adeebshihadeh committed:
add small minimum time between taps to onboarding (#23212)

Co-authored-by: Comma Device <device@comma.ai>

adeebshihadeh committed:
androidd: reduce cloudlog spam

adeebshihadeh committed:
update contributing.md

mitchellgoffpc committed:
Updated release notes

adeebshihadeh committed:
bump version to 0.8.13

deanlee committed:
move LayoutWidget,ClickableWidget to controls (#23207)

deanlee committed:
UI: add timeout to close settings window while onroad (#22980)

* refactor updateWakefulness

* don't hide sidebar

* rename reset->reset_timeout

* move related variables and functions into class Device

* reset timeout in ctor

* set timeout to 10s if ignition is on

* cleanup

* remove unused QTimer *timer

* check getTimestamp

* keep socket reading in update_state

* keep filtering

Co-authored-by: Willem Melching <willem.melching@gmail.com>

deanlee committed:
C2: display ip address in network panel (#23215)

* display ipv4 address

* refactor to class

* only display ip address

sshane committed:
version.py: make function names more clear (#23216)

* rename a few version functions

* and is_prebuilt

* and some formatting 😊

DevinLeamy committed:
athena: persist upload queue to disk (#22659)

* revert submodules

* lowercase

* addressed comments

* add test for cancelled uploads

* formatting

* catch all exceptions

* handle empty param

Co-authored-by: Willem Melching <willem.melching@gmail.com>

pd0wm committed:
comma two: remove lens sag compensation (#23113)

deanlee committed:
put logcat in bootlog (#23217)

* put logcat in boot log

* continue

* use std::unique_ptr

* Revert "use std::unique_ptr"

This reverts commit c671f8ca7160691791bade508c4a8eb5cdb73fb8.

sshane committed:
Remove minSpeedCan (#22991)

* Remove minCanSpeed

Remove minCanSpeed

* it actually only goes out to 2.5 seconds, this is okay to remove

* test to see if this preserves behavior

add minSpeedCan

* Revert "test to see if this preserves behavior"

This reverts commit 31b11f017f7e9da7654fc8064b5983d4a6cc22e5.

* preserve behavior (don't enter stopping as early)

* vEgoStopping needs to be less than or equal to vEgoStarting to avoid state oscillation

ZwX1616 committed:
DM: more adaptive pose policy (#23184)

* rename and add dep

* proto thresholds

* tweak vk

* update natural offset and clip offsets

* 95th looks good

* no punish for being relaxed

* yaw laplace only

* some pay more attention

* update ref

Co-authored-by: Willem Melching <willem.melching@gmail.com>

sshane committed:
seems to be flipped (#23213)

haraschax committed:
Update ref with minSpeedCAN removed

haraschax committed:
Dont use erratic current block (#23211)

haraschax committed:
Revert "Dont use erratic current block (#23211)" (#23221)

This reverts commit 19335bb6a48f0e3558c837557a1ae7c66b8b8587.

haraschax committed:
Calibrationd: don't use noisy current block (#23222)

haraschax committed:
Calibrationd: check size of valid_blocks without current (#23223)

krkeegan committed:
Toyota: Display Blue Barriers When Engaged, Matches OEM Design (#23150)

adeebshihadeh committed:
modeld: start publishing before calib seen (#23190)

adeebshihadeh committed:
cloudlog: add daemon name to context for python daemons (#23225)

* python

* manager is a daemon too

* Update selfdrive/manager/process.py

Co-authored-by: Comma Device <device@comma.ai>

martinl committed:
Subaru Pre-Global: Rename ES_CruiseThrottle to ES_Distance (#23024)

* Rename preglobal ES_CruiseThrottle to ES_Distance

* bump opendbc

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

deanlee committed:
ui: singleton uistate (#22789)

* singleton uistate

* merge master

* merge master

* merge master

* connect uistate in sidebar

* move to device

adeebshihadeh committed:
selfdrive/version: remove one unnecessary write to .git/ (#23183)

deanlee committed:
qt/util: new function scaledPixmap (#23103)

* new function scaledPixmap

* rename to loadpixmap

deanlee committed:
driving.h: remove obsolete poly_fit function prototype (#23229)

deanlee committed:
modeld: make valid variable thread safe (#23228)

pd0wm committed:
manage_athenad.py: add daemon argument to launcher

adeebshihadeh committed:
Revert "Qt/SetingWindow: faster initialization and painting (#21764)"

This reverts commit f482354e4e2ce5d96f5b0c96f7ba923611dc199d.

deanlee committed:
UI: always clear CameraView background (#23231)

* fix background

* gate that with Hardware::EON

* always call glclear

jyoung8607 committed:
VW MQB: Add FW for 2018 Škoda Superb (#23233)

adeebshihadeh committed:
soundd: back to onroad only (#23236)

martinl committed:
Subaru: Add Impreza/Crosstrek 2020 (#21011)

* Add Subaru Impreza/Crosstrek 2020

* Update selfdrive/test/test_routes.py

Co-authored-by: Adeeb Shihadeh <adeebshihadeh@gmail.com>

adeebshihadeh committed:
model replay: more consistent replay (#23237)

* first cleanup

* don't send calib and desire

* should always be valid

* little more

deanlee committed:
framereader: use the same alignement as visionbuf (#23208)

* aligned rgb buffer

* fix yuv stride

adeebshihadeh committed:
boardd: add controls heartbeat (#23185)

* boardd: add controls heartbeat

* bump panda

* bump again

* bump panda

pd0wm committed:
Log actuators after applying rate limits in CarController (#23230)

* return actuators from carcontroller

* log it

* pass to latcontrol

* chrysler

* gm

* honda

* more brands

* rest of the brands

* gm cleanup

* hyundai cleanup

* update ref

* rename field

* fix subaru

* add types

* more subaru fixes #23240

* consistent whitespace

* bump cereal

EwoutH committed:
Convert format strings strings to f-strings (#23241)

* Convert all text strings to f-strings

Reformats all the text from the old "%-formatted" and .format(...) format to the newer f-string format, as defined in PEP 498. This requires Python 3.6+.

Flynt 0.69 was used to reformat the strings. 120 f-strings were created in 51 files.

F-strings are in general more readable, concise and performant. See also: https://www.python.org/dev/peps/pep-0498/#rationale

* revert pyextra changes

* revert ublox.py

Co-authored-by: Willem Melching <willem.melching@gmail.com>

pd0wm committed:
Abstract classes inherit from ABC (#23246)

adeebshihadeh committed:
update release notes

deanlee committed:
modeldata: use a constexpr function to build the IDXS (#23245)

remove convert_array_to_type

rename to build_idxs

space

double

adeebshihadeh committed:
update model replay ref commit after increased model idx precision

deanlee committed:
boardd: new class USBDevice (#23015)

* USBDevice

* merge master

* rebase master

* add to fiels_common

* inline functions

* namespace

* init err

* remove nested if

* cleanup

* bump panda for branch

* change includes

* bump panda branch

* change name

* bump master panda

Co-authored-by: Igor Biletskyy <briskspirit@users.noreply.github.com>
Co-authored-by: Igor Biletskyy <bs@privacy.im>

briskspirit committed:
Revert "boardd: new class USBDevice (#23015)"

This reverts commit 4061f50da3dabff7a16e2a845d66efe00f9d7978.

haraschax committed:
Road Roll Compensation Rebased (#23251)

* first commit

* update refs

sunnyhaibin committed:
Merge branch 'master-niro-ev-2022-port-1' of https://github.com/sunnyhaibin/openpilot-1 into master-niro-ev-2022-port-1

sunnyhaibin commented:
Closing this PR as a car port is not required for the 2022 Kia Niro EV. The issue is caused by an incorrect firmware `esp` from a different car that was added to `KIA_NIRO_EV` during a previous firmware dump. New PR can be found here: #23253.
